TcUnit-Runner/VisualStudioInstance.cs

```diff
@@ -127,35 +131,61 @@ private void LoadDevelopmentToolsEnvironment(string visualStudioVersion, bool is
             tcAutomationSettings.SilentMode = true; // Only available from TC3.1.4020.0 and above
 
             // Load the correct version of TwinCAT using the remote manager in the automation interface
-            log.Info("Using the TwinCAT remote manager to load TwinCAT version '" + this.tcVersion + "'...");
             ITcRemoteManager remoteManager = dte.GetObject("TcRemoteManager");
-            
-            // If the version is pinned check if the pinned version is available
-            if (isTcVersionPinned)
+
+            string latestTcVersion = "";
+            // Check if Version is installed
+            foreach (var possibleVersion in remoteManager.Versions)
             {
-                log.Info("Project has a pinned version: " + this.tcVersion);
-                foreach (var possibleVersion in remoteManager.Versions)
+                latestTcVersion = Char.ToString(possibleVersion[0]);
+                if (possibleVersion == this.tcVersion)
                 {
-                    if (possibleVersion == this.tcVersion)
-                    {
-                        isVersionAvailable = true;
-                        break;
-                    }
+                    isVersionAvailable = true;
                 }
-                if (isVersionAvailable)
-                    log.Info("The pinned TwinCAT version is available ");
-                else
+                // Check if optional Version is installed
+                if (possibleVersion == this.forceToThisTwinCatVersion && this.forceToThisTwinCatVersion != null)
                 {
-                    log.Error("The pinned TwinCAT version is not available ");
-                    throw new Exception("The pinned TwinCAT version is not available");
+                    isForceVersionAvailable = true;
+                    log.Info("The optional TC Version is available ");
```

Comment:
I think it's more clear if the text says TwinCAT and not TC

Reply:
This is applicable for many places in the code

Reply:
Should I change TC by TwinCat? that would generate some changes now. I would make a separate issue for this " Replace TC to TwinCat".